src/libPMacc/include/fields/operations/AddExchangeToBorder.hpp

```diff
@@ -0,0 +1,228 @@
+                    DataSpace< dim > targetCell( blockCell + cellIdx );
+                    DataSpace< dim > sourceCell( targetCell - nullSourceCell );
+
+                    bool assignValue = true;
+
+                    for( uint32_t d = 0; d < dim; ++d )
+                    {
+                        if( direction[ d ] == 1 )
+                        {
+                            if( cellIdx[ d ] < SuperCellSize::toRT()[ d ] - exchangeSize[ d ] )
+                                assignValue = false;
+                            sourceCell[ d ] -= SuperCellSize::toRT()[ d ] - exchangeSize[ d ];
+                            targetCell[ d ] -= SuperCellSize::toRT()[ d ];
+                        }
+                        else if( direction[ d ] == -1 )
+                        {
+                            if( cellIdx[ d ] >= exchangeSize[ d ] )
+                                assignValue = false;
+                            targetCell[ d ] += SuperCellSize::toRT()[ d ];
+                        }
+                    }
+                    if( assignValue )
+                        destBox( targetCell ) += exchangeBox( sourceCell );
+                }
+            );
+        }
+    };
+
+
+    /** add a exchange buffer to the border of the local buffer
+     *
+     * CopyGuardToExchange is the contrarious operation for the neighboring
+     * device to create a exchange which can be added with this functor.
+     */
+    struct AddExchangeToBorder
+    {
+        /** add exchange to border of the local buffer
+         *
+         * Add data cell wise from the exchange to the border of the local buffer.
+         * The `template< typename T> operator+( T const & rhs )` must be defined for
+         * the value type of the buffer.
+         *
+         * @tparam T_DestBuffer PMacc::GridBuffer, type of the used buffer
+         * @tparam T_SuperCellSize PMacc::math::CT::vector, size of the supercell in each direction
+         *
+         * @param destBuffer destination buffer with exchanges
+         * @param superCellSize compile time supercell size
+         * @param exchangeType the exchange direction which needs to be copied
+         */
+        template<
+            typename T_DestBuffer,
+            typename T_SuperCellSize
+        >
+        void operator()(
+            T_DestBuffer & destBuffer,
+            T_SuperCellSize const & superCellSize,
+            uint32_t const exchangeType
+        ) const
+        {
+            boost::ignore_unused( superCellSize );
+
+            using SuperCellSize = T_SuperCellSize;
+
+            constexpr int dim = T_SuperCellSize::dim;
+
+            using MappingDesc = MappingDescription<
+                dim,
+                SuperCellSize
+            >;
+
+            /* use only the x dimension to determine the number of supercells in the guard
+             * PMacc restriction: all dimension must have the some number of guarding
```

Comment:
if you don't put a `.` after the last sentence, add a newline or maybe even a `@warning`